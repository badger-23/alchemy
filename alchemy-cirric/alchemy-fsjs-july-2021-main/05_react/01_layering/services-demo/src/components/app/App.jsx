import React from 'react';
import ThisDay from '../../containers/ThisDay';

export default function App() {
  return <ThisDay />;
}
