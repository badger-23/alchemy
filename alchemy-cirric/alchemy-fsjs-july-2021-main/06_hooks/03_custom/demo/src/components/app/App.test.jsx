import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

const server = setupServer(
  rest.get(
    'https://history.muffinlabs.com/date/:month/:day',
    (req, res, ctx) => {
      return res(
        ctx.json({
          data: { Events: [{ year: '1992', text: 'Bill got chickenpox!' }] },
        })
      );
    }
  )
);

describe('Events App', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a list of events for a given day', async () => {
    // default render is to the root "/" path
    const { container } = render(
      <MemoryRouter initialEntries={['/1/12']}>
        <App />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'events' });
    // Behavioral testing an individual component
    expect(ul).not.toBeEmptyDOMElement();

    const li = screen.getByText('1992 - Bill got chickenpox!');
    // Behavioral testing an individual component
    expect(li).toBeInTheDocument();

    // Using a snapshot on the entire App
    expect(container).toMatchSnapshot();
  });

  it('renders an about page on /about', () => {
    render(
      <MemoryRouter initialEntries={['/about']}>
        <App />
      </MemoryRouter>
    );

    screen.getByText('About Page');
  });
});
