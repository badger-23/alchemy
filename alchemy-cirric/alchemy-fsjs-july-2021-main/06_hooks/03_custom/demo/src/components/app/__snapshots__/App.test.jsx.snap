// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events App renders a list of events for a given day 1`] = `
<div>
  <ul
    aria-label="events"
  >
    <li>
      <article>
        1992
         - 
        Bill got chickenpox!
      </article>
    </li>
  </ul>
</div>
`;
